xarray/tests/test_sparse.py

```diff
@@ -0,0 +1,647 @@
+])
+def test_1d_variable_method(func, sparse_output):
+    var_s = make_xrvar({'x': 10})
+    var_d = xr.Variable(var_s.dims, var_s.data.todense())
+    ret_s = func(var_s)
+    ret_d = func(var_d)
+
+    if sparse_output:
+        assert isinstance(ret_s.data, SparseArray)
+        assert np.allclose(ret_s.data.todense(), ret_d.data)
+    else:
+        assert np.allclose(ret_s, ret_d)
+
+
+class TestSparseVariable:
+    @pytest.fixture(autouse=True)
+    def setUp(self):
+        self.data = sparse.random((4, 6), random_state=0, density=0.5)
+        self.var = xr.Variable(('x', 'y'), self.data)
+
+    def test_unary_op(self):
+        assert_sparse_eq(-self.var.data, -self.data)
+        assert_sparse_eq(abs(self.var).data, abs(self.data))
+        assert_sparse_eq(self.var.round().data, self.data.round())
+
+    def test_binary_op(self):
+        assert_sparse_eq((2 * self.var).data, 2 * self.data)
+        assert_sparse_eq((self.var + self.var).data, self.data + self.data)
+        # assert_eq((self.var[0] + self.var).data, self.data[0] + self.data)
+
+    def test_univariate_ufunc(self):
+        assert_sparse_eq(np.sin(self.data), xu.sin(self.var).data)
+
+    def test_bivariate_ufunc(self):
+        assert_sparse_eq(np.maximum(self.data, 0),
+                         xu.maximum(self.var, 0).data)
+        assert_sparse_eq(np.maximum(self.data, 0),
+                         xu.maximum(0, self.var).data)
+
+    def test_repr(self):
+        expected = dedent("""\
+        <xarray.Variable (x: 4, y: 6)>
+        <COO: shape=(4, 6), dtype=float64, nnz=12, fill_value=0.0>""")
+        assert expected == repr(self.var)
+
+    def test_pickle(self):
+        v1 = self.var
+        v2 = pickle.loads(pickle.dumps(v1))
+        assert_sparse_eq(v1.data, v2.data)
+
+    @pytest.mark.xfail(reason="Missing implementation for np.result_type")
+    def test_missing_values(self):
+        a = np.array([0, 1, np.nan, 3])
+        s = COO.from_numpy(a)
+        var_s = Variable('x', s)
+        assert np.all(var_s.fillna(2).data.todense() == np.arange(4))
+        assert np.all(var_s.count() == 3)
+
+
+@pytest.mark.parametrize("prop", [
+    'attrs',
+    'chunks',
+    'coords',
+    'data',
+    'dims',
+    'dtype',
+    'encoding',
+    'imag',
+    'indexes',
+    'loc',
+    'name',
+    'nbytes',
+    'ndim',
+    'plot',
+    'real',
+    'shape',
+    'size',
+    'sizes',
+    'str',
+    'variable',
+])
+def test_dataarray_property(prop):
+    arr = make_xrarray({'x': 10, 'y': 5})
+    getattr(arr, prop)
+
+
+@pytest.mark.parametrize("func,sparse_output", [
+    (do('all'), False),
+    (do('any'), False),
+    (do('assign_attrs', {'foo': 'bar'}), True),
+    (do('assign_coords', x=make_xrarray({'x': 10}).x + 1), True),
+    (do('astype', int), True),
+    (do('broadcast_equals', make_xrarray({'x': 10, 'y': 5})), False),
+    (do('clip', min=0, max=1), True),
+    # (do('close'), False),
```

Comment:
I'm guessing the commented out cases don't work yet? :)

Reply:
Oops, forgot to finish those! :) 
I've added tests for all except those I couldn't figure out how to use like `swap_dims`, `sel_points` and `isel_points`.